packages/kbn-config-schema/src/internals/index.ts

```diff
@@ -82,7 +82,23 @@ export const internals = Joi.extend([
     base: Joi.boolean(),
     coerce(value: any, state: State, options: ValidationOptions) {
       // If value isn't defined, let Joi handle default value if it's defined.
-      if (value !== undefined && typeof value !== 'boolean') {
+      if (value === undefined) {
+        return value;
+      }
+
+      // Allow strings 'true' and 'false' to be coerced to booleans (case-insensitive).
+
+      // From Joi docs on `Joi.boolean`:
+      // > Generates a schema object that matches a boolean data type. Can also
+      // >  be called via bool(). If the validation convert option is on
+      // > (enabled by default), a string (either "true" or "false") will be
+      // converted to a boolean if specified.
+      if (typeof value === 'string') {
+        const normalized = value.toLowerCase();
+        value = normalized === 'true' ? true : normalized === 'false' ? false : value;
```

Comment:
Do we really want the `toLowerCase`? if the main reason for the change is the query param issue where bools are received as strings in the request, they will always be lowercased anyway. And regarding yaml configuration, I'm not sure we should accept things like `myValue: TrUe` ? 

But I'm fine either way.

Reply:
Ideally I'd stick to only `^true|True|TRUE|false|False|FALSE$` regex to be a subset of [allowed yaml bool values](https://yaml.org/type/bool.html), but current solution works for me as well.

Reply:
I agree that accepting `TrUe` is a bit 🤢, but I'll leave it as-is, only because it matches the Joi default behavior and that is the least likely to break any Kibana installations.

Will update the Readme, thanks.